src/Datadog.Trace/Util/ThreadStaticRandom.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+namespace Datadog.Trace.Util
+{
+    internal class ThreadStaticRandom
```

Comment:
I'm not a big fan of the name to be honest. I feel like it exposes too much implementation detail. What about RandomFactory?

Or a completely different approach: today we only ever use random for generating an id. I don't really like that we need to use two different classes located in two different namespaces for such a simple task. How about just introducing a static `IdGenerator` class, and completely encapsulate the RNG management inside of it?

Reply:
Pushed a new commit demonstrating this approach. I'll keep whichever you prefer

Reply:
I like it.